pandas/compat/_optional.py

```diff
@@ -72,6 +72,119 @@
     "tables": "pytables",
 }
 
+# Mapping of operation contexts to alternative dependencies
+OPERATION_CONTEXTS = {
+    "excel": {
+        "alternatives": ["openpyxl", "xlsxwriter", "calamine", "xlrd", "pyxlsb", "odfpy"],
+        "description": "Excel file operations",
+    },
+    "plotting": {
+        "alternatives": ["matplotlib"],
+        "description": "plotting operations",
+        "fallback": "Use df.describe() for text-based data summaries",
+    },
+    "html": {
+        "alternatives": ["lxml", "html5lib", "beautifulsoup4"],
+        "description": "HTML parsing",
+    },
+    "xml": {
+        "alternatives": ["lxml"],
+        "description": "XML parsing", 
+    },
+    "sql": {
+        "alternatives": ["sqlalchemy", "psycopg2", "pymysql"],
+        "description": "SQL database operations",
+    },
+    "performance": {
+        "alternatives": ["numexpr", "bottleneck", "numba"],
```

Comment:
These are "alternatives" but only in a very loose sense. They are not alternatives for the same operations.